src/v3/src/components/Widget/index.tsx

```diff
@@ -67,12 +70,22 @@ export const Widget: FunctionComponent<WidgetProps> = (widgetProps) => {
 
   const TERMINAL_STATUSES = [IdxStatus.TERMINAL, IdxStatus.SUCCESS];
   const [data, setData] = useState({});
-  const [messages, setMessages] = useState<IdxMessage[]>([]);
+  const [message, setMessage] = useState<IdxMessage | undefined>();
   const [idxTransaction, setIdxTransaction] = useState<IdxTransaction | undefined>();
   const [authApiError, setAuthApiError] = useState<AuthApiError>();
   const pollingTransaction = usePolling(idxTransaction, widgetProps, data);
   const [stepperStepIndex, setStepperStepIndex] = useState<number>(0);
 
+  useEffect(() => {
+    // If we need to load a language (or apply custom i18n overrides), do
+    // this now and re-run render after it's finished.
+    if (!Bundles.isLoaded(getLanguageCode(widgetProps))) {
+      (async () => {
+        await loadLanguage(widgetProps);
+     })();
+    }
+  }, []);
```

Comment:
should we watch on languageCode here to re-render when it changes?

Reply:
also, how widget is re-rendered based on the new properties file?

Reply:
Languages can't be changed midsession, the config properties that controls this is injected into the widget when the `OktaSignin` module is instantiated.